packages/transformer/src/test/standalone/IModelTransformer.test.ts

```diff
@@ -642,6 +645,55 @@ describe("IModelTransformer", () => {
     iModelShared.close();
   });
 
+  it("should log unresolved references", async () => {
+    const iModelShared: SnapshotDb = IModelTransformerTestUtils.createSharedIModel(outputDir, ["A", "B"]);
+    const iModelA: SnapshotDb = IModelTransformerTestUtils.createTeamIModel(outputDir, "A", Point3d.create(0, 0, 0), ColorDef.green);
+    IModelTransformerTestUtils.assertTeamIModelContents(iModelA, "A");
+    const iModelExporterA = new IModelExporter(iModelA);
+
+    // Exclude element
+    const excludedId = iModelA.elements.queryElementIdByCode(Subject.createCode(iModelA, IModel.rootSubjectId, "Context"));
+    assert.isDefined (excludedId);
+    iModelExporterA.excludeElement(excludedId!);
+
+    const subjectId: Id64String = IModelTransformerTestUtils.querySubjectId(iModelShared, "A");
+    const transformerA2S = new IModelTransformer(iModelExporterA, iModelShared, { targetScopeElementId: subjectId });
+    transformerA2S.context.remapElement(IModel.rootSubjectId, subjectId);
+
+    // Configure logger to capture warning message about unresolved references
+    const messageStart = "The following elements were never fully resolved:\n";
+    const messageEnd = "\nThis indicates that either some references were excluded from the transformation\nor the source has dangling references.";
+
+    let unresolvedElementMessage: string | undefined;
+    const logWarning = (_category: string, message: string, _metaData: LoggingMetaData) => {
+      if (message.startsWith (messageStart)) {
+        unresolvedElementMessage = message;
+      }
+    };
+    Logger.initialize(undefined, logWarning);
+    Logger.setLevelDefault(LogLevel.Warning);
+
+    // Act
+    await transformerA2S.processAll();
+
+    // Collect expected ids
+    const result = iModelA.queryEntityIds({ from: "BisCore.Element",
+      where: "Model.Id = :rootId AND ECInstanceId NOT IN (:rootId, :excludedId)",
+      bindings: {rootId: IModel.rootSubjectId, excludedId } });
+    const expectedIds = [...result].map ((x) => `e${x}`);
```

Comment:
Can you also assert that there is atleast one expectedId? I don't think the test would pass if there were 0, but just to make that explicit. You can do that when you get write perms to make a branch if you'd like.

Reply:
Added assert in new PR: https://github.com/iTwin/imodel-transformer/pull/144. Closing this PR.